src/irmin-pack/IO.mli

```diff
@@ -45,9 +45,10 @@ end
 module Unix : S
 
 val with_cache :
-  v:(fresh:bool -> shared:bool -> readonly:bool -> string -> 'a) ->
+  v:(aux:'b -> fresh:bool -> shared:bool -> readonly:bool -> string -> 'a) ->
```

Comment:
Somewhat unsatisfying way of getting around the problem of constructor arguments that don't need to be understood by the `with_cache` wrapper: lift inside a closure with polymorphic parameter. Perhaps warrants either a comment or a simpler implementation?